docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+
+# Define a trajectory with 1000 time-points representing the recorded trajectory of the animal
+T = 1000
+x_coord = np.linspace(0, 10, 1000)
+y_coord = np.linspace(0, 50, 1000)
+
+# Evaluate the product basis at the x and y coordinates
+eval_basis = prod_basis.evaluate(x_coord, y_coord)
+
+# Output the number of elements and samples of the evaluated basis, 
+# as well as the number of elements in the original 1D basis objects
+print(f"Product of two 1D splines with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- a_basis had {a_basis._n_basis_funcs} elements\n\t- b_basis had {b_basis._n_basis_funcs} elements.")
+
+# %%
+# 2D Basis Elements Plotting
+# -----------------------
+# To plot a 2D basis set, we evaluate the basis on a grid of points over the basis function domain.
+# We use the `evaluate_on_grid` method of neurostatslib.basis for this.
+
+X, Y, Z = prod_basis.evaluate_on_grid(200, 200)
+
+# Setup a 3D plot
+fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+
+# Plot only the support
+Z[Z == 0] = np.nan
+ax.plot_surface(X, Y, Z[50], cmap="viridis", alpha=0.8)
+ax.plot_surface(X, Y, Z[83], cmap="rainbow", alpha=0.8)
+ax.plot_surface(X, Y, Z[125], cmap="inferno", alpha=0.8)
+plt.title(f"Product basis with {eval_basis.shape[0]} elements")
+plt.show()
+
+# %%
+# Define and Evaluate a 2D Additive Basis
+# ---------------------------------------
+# Just like in the multiplicative case, we can create an additive basis function as the sum of two 1D basis functions.
+# We approximate the response function for multi-dimensional stimuli as the sum of two 1D basis functions.
+# For the 2D case, we can approximate our response function as follows:
+# $$
+# f(x,y) \\approx \alpha_i \cdot a_i(x) + \beta_j \cdot b_j(y)
+# $$
+# Here, we simply add two basis objects, 'a_basis' and 'b_basis', together to get the additive basis.
+
+
+additive_basis = a_basis + b_basis
+
+# Evaluate the basis functions for the given trajectory.
+eval_basis = additive_basis.evaluate(x_coord, y_coord)
+
+print(f"Sum of two 1D splines with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- a_basis had {a_basis._n_basis_funcs} elements\n\t- b_basis had {b_basis._n_basis_funcs} elements.")
+
+# %%
+# Plot the Additive Basis Elements
+# --------------------------
+# To plot a 2D additive basis set, we need to evaluate the basis on a grid of points over the basis function domain.
+# This can be done using the `neurostatslib.basis.evaluate_on_grid` method.
+
+X, Y, Z = additive_basis.evaluate_on_grid(200, 200)
+
+fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+
+# Only plot the support
+Z[Z == 0] = np.nan
+ax.plot_surface(X, Y, Z[5], cmap="viridis", alpha=0.8)
+ax.plot_surface(X, Y, Z[25], cmap="inferno", alpha=0.8)
+plt.title(f"Additive basis with {eval_basis.shape[0]} elements")
```

Comment:
same comments as for the multiplicative basis: explain X,Y,Z; why are you inserting NaNs; why plot those two values; what does shape mean?

Reply:
I added a comment about it. It's a visualization issue, if i don't set nan outside the support it will plot the surface over all the grid and the overlap will make the basis function less visible. I choose non overlapping basis for this reason. The other thing we can do is to plot in different subplots, but I like the fact of visualizing the basis in the same frame